src/App.js

```diff
@@ -1,16 +1,45 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const updateChatData = (updatedEntry) => {
+    const entries = chatData.map((entry) => {
+      if (entry.id === updatedEntry.id) {
+        return updatedEntry;
+      } else {
+        return entry;
+      }
+    });
+
+    setChatData(entries);
+    likeCount();
+  };
+
+  const likeCount = () => {
+    return chatData
+      .map((entry) => {
+        return entry.liked ? 1 : 0;
+      })
+      .reduce((acc, current) => acc + current);
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>Chat Between Vladimir and Estragon</h1>
```

Comment:
How could we dynamically get the names of the chat participants? What if the chat were just starting and there weren't any messages yet?